app/models/ext_management_system.rb

```diff
@@ -634,7 +634,7 @@ def build_endpoint_by_role(options)
     endpoint = endpoints.detect { |e| e.role == options[:role].to_s }
     # update or create
     if endpoint
-      endpoint.update(options)
+      endpoint.assign_attributes(options)
```

Comment:
yeah it should have said build or assign :-) actually the comment should not be even there :-)

indeed, this is called e.g. for validation